File: Python/Day 5/mini_parking_system.py
from mini_parking_system_function import *

while True:
	print(display_main_menu())
	user_input = input("Make a selection: ")
	match user_input:
		case "1": 
			print("Slot Numbers: ", slot_number, "\n","Car slots:	 ",car_slots)
			user_input = int(input("Select from the available slot numbers: "))
			
			if user_input == slot_number[user_input] and car_slots[user_input] == 0:
				car_slots = add_car(user_input)
				print("Car parked successfully")
			else:
				print("Oops! Slot occupied")
			print(car_slots)
			
		case "2":
			print(car_slots)
			user_input = int(input("Select the slot your car was parked"))
			if user_input == slot_number[user_input] and car_slots[user_input] == 1:
				car_slots = remove_car(user_input)
				print("Car taken successfully")
			else:
				print("Oops! slot is empty")
			print(car_slots)
		case "3":
			print(car_slots)
		case "4": 
			print("Thank you for using our App!")
		case _:
			print("Invalid input!")

File: Java/Day 5/MiniParkingSystem.java
import java.util.Arrays;
import java.util.Scanner;

public class MiniParkingSystem{
	public static void main(String[] args){
		Scanner inputCollector = new Scanner(System.in);

		int[] carPark = new int[20];
		int[] carParkSlotNumber = {1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20};
		
		while (true){
		
		System.out.print(displayMenu());
		System.out.print("Make a selection :");
		int mainMenuInput =  inputCollector.nextInt();
		switch(mainMenuInput){
			case 1 -> {
				System.out.printf(getCarParkAndSlots(), Arrays.toString(carParkSlotNumber), Arrays.toString(carPark));
				System.out.print("Select a parking slot: ");

				int userInput =  inputCollector.nextInt();

				for (int count = 0; count < carPark.length; count++){
					if (userInput == count+1 && carPark[count] == 0){
						carPark[count] = 1;
						System.out.println("Car parked successfully");
						break;
					}

					if (userInput == count+1 && carPark[count] == 1){
						carPark[count] = 1;
						System.out.println("The selected slot is occupied, choose another slot");
						break;
					}
				}
				System.out.printf(getCarParkAndSlots(), Arrays.toString(carParkSlotNumber), Arrays.toString(carPark));


			}// end of case 1
			case 2 -> {
				System.out.printf(getCarParkAndSlots(), Arrays.toString(carParkSlotNumber), Arrays.toString(carPark));
				System.out.print("Select the slot your car was parked: ");

				int userInput =  inputCollector.nextInt();

				for (int count = 0; count < carPark.length; count++){
					if (userInput == count+1 && carPark[count] == 1){
						carPark[count] = 0;
						System.out.println("Car taken sucessfully");
						break;
					}
					else{
						System.out.println("Wrong slot selected, no car was parked here");
						break;
					}
				}
				System.out.printf(getCarParkAndSlots(), Arrays.toString(carParkSlotNumber), Arrays.toString(carPark));




			} // of of case 2
			case 3 -> {
				System.out.printf(getCarParkAndSlots(), Arrays.toString(carParkSlotNumber), Arrays.toString(carPark));
			}
			case 4 -> {break;}
			default-> {System.out.print("Wrong input, select from the options ");}
		}// end of switch case
		} // end of while loop
		
	}

	public static String getCarParkAndSlots(){
	
		String carPark = """
		Slot Numbers : %s
		Parking Slots : %s
		""";
		return carPark;

	}

	public static String displayMenu(){
	
		String mainMenu = """

		Welcome to Semi Parking System
		>> Park your cars with ease <<	

		1. Park your car.
		2. Take your parked car.
		3. Check available slot

		""";
		return mainMenu;

	}
}